src/gui/MainWindow.cpp

```diff
@@ -168,6 +175,17 @@ MainWindow::MainWindow()
     , m_appExiting(false)
 {
     m_ui->setupUi(this);
+    
+    #ifdef WITH_XC_DBUS
+    #if defined(Q_OS_UNIX)
```

Comment:
This probably will trigger some problem on Q_OS_MACOS.
@weslly can you test this PR on a mac? Thanks

Reply:
@TheZ3ro 

![image](https://user-images.githubusercontent.com/107672/35275185-157a1fdc-0026-11e8-81bf-bfd6428c5ffc.png)

Reply:
@weslly Ok thanks, I will fix this.

Reply:
@TheZ3ro Appending ` && !defined(Q_OS_MAC)` on this line seems to solve the compilation issue, but that would mean the `#undef` macro you used earlier had no effect

Reply:
@TheZ3ro maybe adding this on `src/CMakeLists.txt` would be better:

```
if(APPLE OR MINGW)
    set(WITH_XC_DBUS OFF)
endif()
```

Reply:
@weslly pushed a different fix. Now should always work

Reply:
@TheZ3ro works fine now :)